xformers/ops/fmha/__init__.py

```diff
@@ -569,9 +569,9 @@ def merge_attentions(
     concat_path = attn_is_concat and lse_is_concat
     if not concat_path:
         if attn_is_concat:
-            attn_split = cast(torch.Tensor, attn_split).unbind(0)
```

Comment:
Was this change fixing a failure, or was it to make the type checker happy, or something else?

Reply:
this change seems wrong to me

Reply:
The typing is not quite right at the moment - because unbind returns a tuple and the code which uses it needs a list I think, although it just needs a Sequence.

This change will /work/ because a Tensor is usable as a Sequence of tensors, which is what the non-concat path needs, so the unbind was unnecessary. And this change probably also fixes the type error because mypy doesn't get finer than knowing attn_split is a Union. But this change does make the code slightly less understandable. 

I think OP is just trying to fix the existing lint errors in this PR: here and in the `predicated...h` file (which is clearly the correct fix). I'm fine if they leave them broken or just this one broken.

Reply:
Thanks @bottler. Yes, the changes here are to fix lint error.